contracts/bridges/FxGovernorTunnel.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+// IFxMessageProcessor represents interface to process message
+interface IFxMessageProcessor {
+    function processMessageFromRoot(uint256 stateId, address rootMessageSender, bytes memory data) external;
+}
+
+/// @dev Provided zero address.
+error ZeroAddress();
+
+/// @dev Only self contract is allowed to call the function.
+/// @param sender Sender address.
+/// @param instance Required contract instance address.
+error SelfCallOnly(address sender, address instance);
+
+/// @dev Only `fxChild` is allowed to call the function.
+/// @param sender Sender address.
+/// @param fxChild Required Fx Child address.
+error FxChildOnly(address sender, address fxChild);
+
+/// @dev Only on behalf of `rootGovernor` the function is allowed to process the data.
+/// @param sender Sender address.
+/// @param rootGovernor Required Root Governor address.
+error RootGovernorOnly(address sender, address rootGovernor);
+
+/// @dev Provided incorrect data length.
+/// @param expected Expected minimum data length.
+/// @param provided Provided data length.
+error IncorrectDataLength(uint256 expected, uint256 provided);
+
+/// @dev Target execution failed.
+/// @param target Target address.
+/// @param value Provided value.
+/// @param payload Provided payload.
+error TargetExecFailed(address target, uint256 value, bytes payload);
+
+/// @title FxGovernorTunnel - Smart contract for the governor child tunnel bridge implementation
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author AL
+contract FxGovernorTunnel is IFxMessageProcessor {
```

Comment:
A new contract to establish a connection between the mainnet governor and deployed contracts on L2

Reply:
This contract will be the owner of all the L2 contracts, as it is governed directly by the Timelock across the bridge.